common/db_test.go

```diff
@@ -656,6 +659,83 @@ func TestLoadWinningTickets_GivenTwoSessionsWithTickets_ReturnsAllTickets(t *tes
 	assert.Equal(recipientRand1, recipientRands[1])
 }
 
+func TestInsertMiniHeader_ReturnsFindLatestMiniHeader(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	assert := assert.New(t)
+	require := require.New(t)
+	require.Nil(err)
+
+	h0 := defaultMiniHeader(t)
+	err = dbh.InsertMiniHeader(h0)
+	require.Nil(err)
+
+	h0db, err := dbh.FindLatestMiniHeader()
+	require.Nil(err)
+	assert.Equal(*h0, *h0db)
+
+	// Test FindLatestMiniHeader with 2 blocks
+	h1 := defaultMiniHeader(t)
+	h1.Number = big.NewInt(451)
+	err = dbh.InsertMiniHeader(h1)
+	require.Nil(err)
+	latest, err := dbh.FindLatestMiniHeader()
+	require.Nil(err)
+	assert.Equal(*h1, *latest)
+}
+
+func TestFindAllMiniHeadersSortedByNumber(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	assert := assert.New(t)
+	require := require.New(t)
+	require.Nil(err)
+
+	added := make([]*blockwatch.MiniHeader, 10)
+	for i := 0; i < 10; i++ {
+		h := defaultMiniHeader(t)
+		h.Number = big.NewInt(int64(i))
+		err = dbh.InsertMiniHeader(h)
+		require.Nil(err)
+		added[9-i] = h
+	}
+
+	headers, err := dbh.FindAllMiniHeadersSortedByNumber()
+	for i, h := range headers {
+		assert.Equal(h.Number.Int64(), int64(len(headers)-1-i))
+		assert.Equal(*h, *added[9-i])
+	}
+}
+
+func TestDeleteMiniHeader(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	assert := assert.New(t)
+	require := require.New(t)
+	require.Nil(err)
+
+	h0 := defaultMiniHeader(t)
+	err = dbh.InsertMiniHeader(h0)
+	require.Nil(err)
+
+	h0db, err := dbh.FindLatestMiniHeader()
+	require.Nil(err)
+	assert.Equal(h0, h0db)
+
+	err = dbh.DeleteMiniHeader(h0.Hash)
```

Comment:
Could add a few additional tests:

- When the header doesn't exist
- When there are > 1 headers already in the DB

Reply:
Fixed in ced063e